docs/notebooks/qhsri-tutorial.pct.py

```diff
@@ -0,0 +1,232 @@
+
+plt.scatter(uniform_points[:, 0], uniform_points[:, 1])
+plt.title("Uniformly sampled points")
+plt.show()
+plt.close()
+
+plt.scatter(uniform_pts_mean, uniform_pts_std)
+plt.title("Mean vs negative std of uniformly sampled points")
+plt.xlabel("Mean")
+plt.ylabel("Negative std")
+plt.show()
+
+# %% [markdown]
+# Since we only want the best points in terms of the risk reward tradeoff, we want points that have low means and high standard deviations. So we remove any points that are dominated by another point. A point `a` is dominated by another point `b` if `b.mean` <= `a.mean` and `b.std` >= `a.std`.
+#
+# There is a function in trieste that lets us calculate this non-dominated set. Let's find the non-dominated points and plot them on the above chart.
+
+# %%
+from trieste.acquisition.multi_objective.dominance import non_dominated
+
+uniform_non_dominated = non_dominated(tf.concat([uniform_pts_mean, uniform_pts_std], axis=1))[0]
+
+plt.scatter(uniform_pts_mean, uniform_pts_std)
+plt.scatter(uniform_non_dominated[:, 0], uniform_non_dominated[:, 1], c="r")
+plt.title("Mean vs negative std of uniformly sampled points")
+plt.xlabel("Mean")
+plt.ylabel("Negative std")
+plt.show()
+
+print(f"There are {len(uniform_non_dominated)} non-dominated points")
+
+# %% [markdown]
+# We can see that there's only a few non-dominated points to choose from the select the next batch. This set of non-dominated points is the Pareto front in the optimisation task of minimising mean and maximising standard deviation.
+#
+# This means we can improve on this by using optimisation rather than random sampling to pick the points that we will select our batch from. In trieste we use the NSGA-II multi-objective optimisation method from the [pymoo](https://pymoo.org/) library to find the Pareto front.
+#
+# The `BatchHypervolumeSharpeRatioIndicator` acqusition rule makes use of the `_MeanStdTradeoff` class, which expresses the optimisation problem in the pymoo framework. Pymoo is then used to run the optimisation.
+#
+# NSGA-II is a genetic algorithm, and so we need to define a population size and number of generations.
+
+# %%
+import numpy as np
+from trieste.acquisition.rule import _MeanStdTradeoff
+from pymoo.algorithms.moo.nsga2 import NSGA2
+from pymoo.optimize import minimize
+
+problem = _MeanStdTradeoff(model, search_space)
+algorithm = NSGA2(pop_size=500)
+result = minimize(problem, algorithm, ("n_gen", 200), seed=1, verbose=False)
+optimised_points, optimised_mean_std = result.X, result.F
+
+# %% [markdown]
+# Now we can plot the points that have been optimised for mean and standard deviation, and their means and stds.
+
+# %%
+plt.scatter(result.X[:, 0], result.X[:, 1])
+plt.title("Points optimised for mean and std")
+plt.show()
+
+plt.scatter(result.F[:, 0], result.F[:, 1])
+plt.title("Mean vs std of optimised points")
+plt.show()
+
+# %% [markdown]
+# We can check the non-dominated points again, and see that this is a great improvement.
```

Comment:
... , and see that the outcome of NSGA-II is much better than the randomly sampled ones.

(assuming this is the case)